services/accountConfirmation.service.js

```diff
@@ -0,0 +1,117 @@
+"use strict";
+const logger = require("./logger.service");
+const AccountConfirmation = require("../models/accountConfirmationToken.model");
+const Constants = require("../constants");
+const jwt = require("jsonwebtoken");
+
+/**
+ * @function findByAccountId
+ * @param {ObjectId} accountId
+ * @return {DocumentQuery} The document query will resolve to either account confirmation or null.
+ * @description Finds an account confirmation document by accountId.
+ */
+function findByAccountId (accountId) {
+    const TAG = `[ AccountConfirmation Reset Service # findByAccountId ]`;
+    return AccountConfirmation.findOne({
+        accountId: accountId
+    }, logger.queryCallbackFactory(TAG, "AccountConfirmation", "accountId"));
+}
+
+/**
+ * @function findById
+ * @param {ObjectId} id
+ * @return {DocumentQuery} The document query will resolve to either account confirmation or null.
+ * @description Finds an account by mongoID.
+ */
+function findById (id) {
+    const TAG = `[ AccountConfirmation Service # findById ]`;
+    return AccountConfirmation.findById(id, logger.queryCallbackFactory(TAG, "AccountConfirmation", "mongoId"));
+}
+
+/**
+ * Creates Account Confirmation document in the database
+ * @param {String} type the type of user which to create the token for
+ * @param {String} email
+ * @param {ObjectId} accountId optional accountId parameter to link to account, optional when token is being made for not a hacker
+ * @returns {Promise.<*>}
+ */
+async function create (type, email, accountId) {
+    //Create new instance of account confirmation
+    const newAccountToken = AccountConfirmation({
+        accountType: type,
+        email: email
+    });
+    if(accountId !== undefined){
+        newAccountToken.accountId = accountId;
+    }
+    return newAccountToken.save();
+}
+
+/**
+ * Generates JWT for Confirming account
+ * @param {ObjectId} accountConfirmationId
+ * @param {ObjectId} accountId
+ * @returns {string} JWT Token containing accountId and accountConfirmationId
+ */
+function generateToken (accountConfirmationId, accountId) {
+    const token = jwt.sign({
+        accountConfirmationId: accountConfirmationId,
+        accountId: accountId
+    }, process.env.JWT_CONFIRM_ACC_SECRET, {
+        expiresIn: "7 day"
```

Comment:
What happens when a user tries to use a token that has already expired? How can they get a new token for their account?

Reply:
Will be handled in another P.R. as an issue, based it off of the resetPassword token but realized that doesn't handle expiration either. ATM an expired token will still work but I think this will be more time sensitive issue when there are tokens floating around to make admin accounts(I'll handle those two in one PR)

Reply:
Sounds good.